java/semantickernel-api/src/main/java/com/microsoft/semantickernel/orchestration/ToolCallBehavior.java

```diff
@@ -135,24 +128,26 @@ public boolean functionEnabled(String pluginName, String functionName) {
         return functionSettings.isEmpty();
     }
 
-    public int maximumAutoInvokeAttempts() {
-        return getSetting("maximumAutoInvokeAttempts", DEFAULT_MAXIMUM_AUTO_INVOKE_ATTEMPTS);
-    }
-
-    protected void setFlag(String key, boolean value) {
-        flags.put(key, value);
-    }
-
-    protected void setSetting(String key, int value) {
-        settings.put(key, value);
+    public int getMaximumAutoInvokeAttempts() {
+        return this.maximumAutoInvokeAttempts;
     }
+    public void configureOptions(
+            ChatCompletionsOptions options,
+            List<FunctionDefinition> functions) {
+        if (functions.isEmpty()) {
+            return;
+        }
 
-    protected boolean getFlag(String key) {
-        return flags.getOrDefault(key, false);
-    }
+        options.setTools(functions.stream()
+//            .filter(function -> {
```

Comment:
At some point you're going to put this back?

Reply:
Yes, or update it somehow.